### PR TITLE
TTS customisation guide

### PR DESCRIPTION
**Pull request by AshishSardana**

### CODE REVIEW

File: tts-how-to-customize-speech.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+[HifiGAN](https://arxiv.org/pdf/2010.05646.pdf) is a neural vocoder model for text-to-speech applications. It is intended as the second part of a two-stage speech synthesis pipeline, with a mel-spectrogram generator such as FastPitch as the first stage. HifiGAN is a neural vocoder based on a generative adversarial network framework. During training, the model uses a powerful discriminator consisting of small sub-discriminators, each one focusing on specific periodic parts of a raw waveform. The generator is very fast and has a small footprint, while producing high quality speech. 
+
+This pipeline forms a text-to-speech system that enables us to synthesize natural sounding speech from raw transcripts without any additional information such as patterns or rhythms of speech.
+
+The below flow diagram shows the Riva speech synthesis pipeline along with the possible customizations.
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:21]:
```suggestion
The following flow diagram shows the Riva speech synthesis pipeline along with the possible customizations.

```

File: tts-how-to-customize-speech.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+![RIVA TTS pipeline](./imgs/riva-tts-customizations.png "RIVA TTS Pipeline")
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:23]:
```suggestion
![Riva TTS pipeline](./imgs/riva-tts-customizations.png "Riva TTS Pipeline")

```

File: tts-how-to-customize-speech.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The following customizations can help to control and improve the synthesized speech. These customizations are listed in increasing order of difficulty and efforts.:
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:25]:
```suggestion
The following customizations can help control and improve the synthesized speech. These customizations are listed in increasing order of difficulty and efforts:

```

File: tts-how-to-customize-speech.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+| 1. SSML tags                    | Quick and easy  | Changes the pitch, rate, and pronunciation through phoneme.           | When you want to emphasize or want particular pronounciation for certain words | [Customization with SSML]( https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-customization-with-ssml.ipynb)                           |
+| 2. Training FastPitch and HiFiGAN  | Moderately hard | Train the FastPitch and HifiGAN models on custom data | When you want to change the accent, language or voice of the speaker           | [FastPitch/HiFiGAN training](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-training.ipynb), [FastPitch/HiFiGAN deployment](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-deployment.ipynb), with TAO Toolkit |
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:30]:
```suggestion
| 2. Training FastPitch and HiFi-GAN  | Moderately hard | Train the FastPitch and HiFi-GAN models on custom data | When you want to change the accent, language or voice of the speaker           | [FastPitch/HiFi-GAN training](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-training.ipynb), [FastPitch/HiFi-GAN deployment](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-deployment.ipynb), with TAO Toolkit |

```

File: tts-how-to-customize-speech.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+| Techniques                        | Difficulty      | What it does                                              | When to use                                                                    | How to use                                                                                                                                                                                 |
+|-----------------------------------|-----------------|-----------------------------------------------------------|--------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1. SSML tags                    | Quick and easy  | Changes the pitch, rate, and pronunciation through phoneme.           | When you want to emphasize or want particular pronounciation for certain words | [Customization with SSML]( https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-customization-with-ssml.ipynb)                           |
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:29]:
```suggestion
| 1. SSML tags                    | Quick and easy  | Changes the pitch, rate, and pronunciation through phoneme.           | When you want to emphasize or want a particular pronunciation for certain words | [Customization with SSML](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-customization-with-ssml.ipynb)                           |

```

File: tts-how-to-customize-speech.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+| 2. Training FastPitch and HiFiGAN  | Moderately hard | Train the FastPitch and HifiGAN models on custom data | When you want to change the accent, language or voice of the speaker           | [FastPitch/HiFiGAN training](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-training.ipynb), [FastPitch/HiFiGAN deployment](https://github.com/nvidia-riva/tutorials/blob/dev/22.05/tts-python-advanced-pretrain-tts-tao-deployment.ipynb), with TAO Toolkit |
+|                                   |                 |                                                           |                                                                                |                                                                                                                                                                                            |
+
+In the next section, we will give a more detailed discussions of each technique. For a how-to step-by-step guide, consult the notebooks linked in the table.
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:33]:
```suggestion
In the next section, we provide a more detailed discussion of each technique. For a how-to step-by-step guide, consult the tutorials linked in the table.

```

File: tts-how-to-customize-speech.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+- The `prosody` tag, which supports two attributes `rate` and `pitch`, through which we can control the rate and pitch of the generated audio.
+    1. Pitch attribute - 
+    Riva supports an additive relative change to the pitch. The pitch attribute has a range of [-3, 3]. This value returns a pitch shift of the attribute value multiplied with the speaker’s pitch standard deviation when the FastPitch model is trained. 
+
+    2. Rate attribute -
+    Riva supports a percentage relative change to the rate. The rate attribute has a range of [25%, 250%].
+
+- The `phoneme` tag, which allows us to control the pronunciation of the generated audio.
+    1. Phoneme tag -
+    For a given word or sequence of words, use the ph attribute to provide an explicit pronunciation, and the alphabet attribute to provide the phone set. Currently, only x-arpabet is supported for pronunciation dictionaries based on CMUdict.
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:49]:
```suggestion
    For a given word or sequence of words, use the `ph` attribute to provide an explicit pronunciation, and the alphabet attribute to provide the phone set. Currently, only x-arpabet is supported for pronunciation dictionaries based on CMUdict.

```

File: tts-how-to-customize-speech.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+## 2. Training FastPitch and HiFiGAN
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:51]:
```suggestion
## 2. Training FastPitch and HiFi-GAN

```

File: tts-how-to-customize-speech.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+End-to-end training of TTS models requires large datasets and heavy compute resources. 
+
+For this reason, we only recommend training models from scratch where several hundreds of hours of transcribed speech data is available.
+
+While you collect the text dataset for TTS, you need to remember that TTS models learn to map n-grams to sounds. Thus, you should ensure that the text data isn't archaic and it should have sufficient phoneme coverage.
+
+**Note:** Finetuning TTS models is a [recent advancement](https://paarthneekhara.github.io/tlfortts/) in the field of audio synthesis and is usually used to adapt the pre-trained TTS models for a different accent.
```
Comment (LynseyFabel) [inline tts-how-to-customize-speech.md:59]:
```suggestion
**Note:** Fine-tuning TTS models is a [recent advancement](https://paarthneekhara.github.io/tlfortts/) in the field of audio synthesis and is usually used to adapt the pre-trained TTS models for a different accent.

```

**Review by LynseyFabel — APPROVED** [review]
I've finished my review and left a few comments. Thank you.

File: tts-how-to-customize-speech.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+NVIDIA offers pre-trained [FastPitch](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/tao/models/speechsynthesis_hifigan) and [HiFiGAN](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/tao/models/speechsynthesis_hifigan) models, trained on 25+ hours of LJSpeech dataset, which can be used for finetuning. Though finetuning is out of scope for this lab, we would be covering it in the next release. Stay tuned!
```
[the remark this answers is not in the document]
Comment (AshishSardana) [reply tts-how-to-customize-speech.md]: updated to FastPitch link.

**Comment by AshishSardana** [comment]
@LynseyFabel - Thank you so much for the review!
@NikhilSrihari - Are we ready to merge?

**Pull request closed without merging**